src/main/java/com/example/perkmanager/Membership.java

```diff
@@ -0,0 +1,63 @@
+package com.example.perkmanager;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+
+/**
+ * Represents a membership used for perks.
+ */
+@Entity
+public class Membership {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
```

Comment:
Using the primitive long for id means it defaults to 0 before persistence. Using Long should allow it to be null before persistence, correctly indicating "unsaved".
```suggestion
    private Long id;
```